Libs/MRML/Core/vtkMRMLNodePropertyMacros.h

```diff
@@ -0,0 +1,249 @@
+/*=auto=========================================================================
+
+  Portions (c) Copyright 2005 Brigham and Women's Hospital (BWH) All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+=========================================================================auto=*/
+
+#ifndef __vtkMRMLNodePropertyMacros_h
+#define __vtkMRMLNodePropertyMacros_h
+
+
+// Macros for writing node properties to XML
+//----------------------------------------------------------------------------
+
+#define vtkMRMLWriteXMLBeginMacro(of) \
+  { \
+  ostream& xmlWriteOutputStream = of;
+
+#define vtkMRMLWriteXMLEndMacro() \
+  }
```

Comment:
I wonder if these two macros are needed: vtkMRMLWriteXMLBeginMacro / vtkMRMLWriteXMLEndMacro

Reply:
Two reasons:
1. Hide xmlWriteOutputStream local variable (a name clash is unlikely, but just to be on the safe side).
2. Future-proofing: this allows us to improve the macros with any additional initializations/cleanups without requiring any change in the MRML nodes that use them.

Reply:
In that case, 👍  to use macros. May be it could be as an implementation details, function could be used.

@finetjul What do you think ?

Reply:
I've created a separate, small pull request for node property setting helper macros: https://github.com/Slicer/Slicer/pull/890